Transaction/DLC_Contract_Execution_Transaction.md

```diff
@@ -0,0 +1,138 @@
+# DLC Contract Execution Transaction
+
+This details the exact format of the on-chain contract execution transaction (CET), which both sides need to agree on to ensure signatures and adaptor signatures are valid.
+
+# Table of Contents
+
+* [Transaction](#transaction)
+  * [CET Outputs](#cet-outputs)
+    * [Offerer Output](#offerer-output)
+    * [Accepter Output](#accepter-output)
+* [Fees](#fees)
+* [Test Vectors](#test-vectors)
+* [References](#references)
+* [Authors](#authors)
+
+# Transaction
+
+* version: 2
+* locktime: `contract_maturity`
+* txin count: 1
+  * `txin[0]` outpoint: `txid` of funding transaction and `output_index` 0
+  * `txin[0]` sequence: 0xFFFFFFFE
+  * `txin[0]` script bytes: 0
+  * `txin[0]` witness: `0 <signature_for_offer_pubkey> <signature_for_accept_pubkey>`
+
+The output script public keys and `contract_maturity` are negotiated in the offer and accept messages.
+
+There will be one CET for every possible outcome, the output values correspond to such an outcome and are negotiated in the offer message.
+
+## CET Outputs
+
+The Offerer/Initiator's output comes first, the Accepter's output comes second.
+
+Note that this will likely change in the future.
+
+If either party receives less than the dust limit of `1000 satoshis` for this outcome, then their output is not produced.
+
+### Offerer Output
+
+This output sends funds won by the offerer corresponding to this CET's outcome to the offerer's final address specified in the offer message.
+
+### Accepter Output
+
+This output sends funds won by the accepter corresponding to this CET's outcome to the accepter's final address specified in the accept message.
+
+# Fees
+
+Fees for the CET are paid for in the Funding Transaction, the following only shows the weight calculation for the largest possible CET.
+
+```
+multi_sig: 71 bytes
+	- OP_2: 1 byte
+	- OP_DATA: 1 byte (offer_pubkey length)
+	- offer_pubkey: 33 bytes
+	- OP_DATA: 1 byte (accept_pubkey length)
+	- accept_pubkey: 33 bytes
+	- OP_2: 1 byte
+	- OP_CHECKMULTISIG: 1 byte
+
+offer_output: 9 + offer_output_script length bytes
```

Comment:
Aren't output scripts normalized and thus we know their length ?

Reply:
No, there is a final address field in both the offer and accept messages which will specify the SPK used, it does not even have to be segwit.

Reply:
p2wpkh is `OP_0 <20-byte-hash>` vs p2wsh is `OP_0 <32-byte-hash>`

taproot will be `OP_1 <32-byte-pub-key>`